DHT.h

```diff
@@ -37,11 +37,28 @@ written by Adafruit Industries
 
 class DHT {
   public:
-   DHT(uint8_t pin, uint8_t type, uint8_t count=6);
+   // Note that count is now ignored as the DHT reading algorithm adjusts itself
+   // based on the speed of the processor.
+   DHT(uint8_t pin, uint8_t type, uint8_t count) {
+     DHT(pin, type);
```

Comment:
This is not the right C++ syntax to delegate constructors.
It does call the constructor indeed, but it just constructs another
instance instead, which is then immediately discarded. See for example
https://en.wikipedia.org/wiki/C%2B%2B11#Object_construction_improvement

On Fri, Mar 4, 2016, 07:23 Gregg Ubben <notifications@github.com> wrote:

> In DHT.h
> <https://github.com/adafruit/DHT-sensor-library/pull/54#discussion_r54982620>
> :
>
> > @@ -37,11 +37,28 @@ written by Adafruit Industries
> >
> >  class DHT {
> >    public:
> > -   DHT(uint8_t pin, uint8_t type, uint8_t count=6);
> > +   // Note that count is now ignored as the DHT reading algorithm adjusts itself
> > +   // based on the speed of the processor.
> > +   DHT(uint8_t pin, uint8_t type, uint8_t count) {
> > +     DHT(pin, type);
>
> Thank you for your feedback.
>
> I am not sure I understand your comment.
>
> The 3 arg constructor DHT(pin, type, count) calls the 2 arg constructor
> with DHT(pin, type) in the header. The 2 arg constructor is the header
> definition for the 2 arg constructor in the DHT.cpp file. I removed the 3
> arg constructor that used to be in the DHT.cpp file.
>
> Does that make sense or am misunderstanding your comment?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adafruit/DHT-sensor-library/pull/54/files#r54982620>.
>